submissions/krylenger/dom-api-task/script.js

```diff
@@ -0,0 +1,189 @@
+    models: ["BMW S1000RR", "Yamaha YZF-R1", "Kawasaki Ninja H2"],
+    avgPrice: "Average price: $13.000",
+  },
+  {
+    name: "SUPERMOTO",
+    img: "./images/supermoto.png",
+    description: [
+      "Supermoto bikes, sometimes called Supermotard, are designed for racing on tracks made up of tarmac, dirt and jump sections. ",
+      'To optimise the bikes for all three they tend to feature wire-spoke wheels, very long travel suspension, great ground clearance, very basic equipment and a noticeable high "beak" as the front fender.',
+      "They also tend to be simple single cylinder engines with large wheels and chunky tyres.",
+    ],
+    models: [
+      "Husqvarna 701 Supermoto",
+      "KTM 690 SMC R",
+      "Ducati Hypermotard 1100 EVO SP",
+    ],
+    avgPrice: "Average price: $10.000",
+  },
+  {
+    name: "OFFROAD",
+    img: "./images/offroad.png",
+    description: [
+      'The hallmarks of bikes specifically designed for off-road work tend to be very light weight, wire-spoked wheels, some form of protection under the engine in case of grounding, engine bars in case it drops, long travel suspension, hardly any screen in case you go over the top and usually "knobbly" tyres for improved grip.',
+      "They are popular for green laning on public byways / un-tarmacked roads.",
+    ],
+    models: ["HONDA CRF450RX", "HUSQVARNA FX450", "KTM 450XC-F"],
+    avgPrice: "Average price: $11.000",
+  },
+  {
+    name: "RETRO",
+    img: "./images/retro.png",
+    description: [
+      "Most of the retro models you can buy today started life as a sporty naked bike which has been modified to look more traditional, but without sacrificing all the modern tech and riding experience you would expect from a new bike.",
+      "They usually feature a single round headlight, simpler rounded clocks, stitched seats, more subtle colour schemes, rounded exhausts, loads of optional shiny bits and often wire spoked wheels or cast wheels designed to look a little bit like spokes from a distance.",
+      "Closely related are modern cafe racers which tend to start life as retro naked bikes but with the addition of low bars, a headlight fairing and pillion seat hump.",
+    ],
+    models: ["Kawasaki Z900RS", "Triumph Speed Twin", "Moto Guzzi V7 III"],
+    avgPrice: "Average price: $10.000",
+  },
+];
+
+const navBtnToggle = document.querySelector(".nav-btn-toggle");
+const nav = document.querySelector(".nav");
+const navUl = document.querySelector(".nav__ul");
+
+const main = document.querySelector(".main");
+const mainHeader = document.querySelector(".main__header");
+const mainImage = document.querySelector(".main__img");
+const mainDescription = document.querySelector(".main__description");
+const mainDescriptionText = document.querySelector(".main__description-text");
+const mainUlBikes = document.querySelector(".main__ul-bikes");
+const mainAvgPrice = document.querySelector(".main__avg-price");
+
+const handleNavBtnToggle = ({ target }) => {
```

Comment:
Oh.. so many DOM operations here... 
Try to decrease a number of classes on your elements. I mean you can use some default styles for opened nav, main and ul elements. If you'll do this, you don't need to handle so many classes.
For example, you can apply styles from the `nav-btn-toggle--nav-opened` class for your `navBtnToggle` by default and just toggle `nav-btn-toggle--nav-closed` 
More info about the `Element.classList.toggle` you can find [here](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList)
Is it clear for you? Let me know if you have problems with this

Reply:
yes, I was thinking about this (to implement with toggle) 
but in this case I will lose one of these smooth transitions, right?
```
.nav--opened {
    width: 200px;
    transition: all 0.3s ease-in-out;
}

.nav--closed {
    width: 0;
    transition: all 0.3s ease-in-out;
}
```

I mean one will be smooth (i.e. when we close nav tav) and the other one (when we open nav) - instant.

Reply:
Got it! 
All we need is just to add transition to .nav, not to modificator!

Reply:
✅ Done
I really appreciate your feedback! 
Every time this helps me to learn new stuff and accomplish things in the right way!

Reply:
Cool! I'm glad that all of our recommendations help others to grow!